### PR TITLE
Merge pull request #5571 from AnanthaC/RDK-39655-red

### PR DESCRIPTION
**Pull request by AnanthaC**
Cherry-pick RDK-39655 changes to the main branch

### CODE REVIEW

File: SystemServices/SystemServices.cpp [hunk 1]
```diff
@@ -2280,6 +2280,23 @@ namespace WPEFramework {
             }
         }
```
Comment (anand-ky) [inline SystemServices/SystemServices.cpp:2282]: Need to also update version in the macro - https://github.com/rdkcentral/rdkservices/blob/63dde5c3124550f01b203bedae553fa32acb8905/SystemServices/SystemServices.cpp#L69

File: SystemServices/CHANGELOG.md [hunk 2]
```diff
@@ -15,6 +15,10 @@ All notable changes to this RDK Service will be documented in this file.
 * Changes in CHANGELOG should be updated when commits are added to the main or release branches. There should be one CHANGELOG entry per JIRA Ticket. This is not enforced on sprint branches since there could be multiple changes for the same JIRA ticket during development. 
 
 * For more details, refer to [versioning](https://github.com/rdkcentral/rdkservices#versioning) section under Main README.
+## [3.1.2] - 2024-09-17
+### Added
```
Comment (anand-ky) [inline SystemServices/CHANGELOG.md:19]: Pls update minor version and reset patch version. So move to "3.2.0". Details on versioning mentioned in the README - https://github.com/rdkcentral/RDKServices?tab=readme-ov-file#versioning

**Review by anand-ky — CHANGES_REQUESTED** [review]

File: SystemServices/CHANGELOG.md (hunk 2 above)
Comment (AnanthaC) [reply SystemServices/CHANGELOG.md:19]: Updated

File: SystemServices/System.json [hunk 3]
```diff
@@ -2023,6 +2034,20 @@
                 ]
             }
         },
+	"onRecoveryStateChange":{
```
Comment (anand-ky) [inline SystemServices/System.json:2037]: We cannot add newline to the json. The md file generator will break. Pls check and get this fixed

**Pull request merged**